resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerAspect.java

```diff
@@ -132,7 +133,21 @@ private CircuitBreaker getCircuitBreakerAnnotation(ProceedingJoinPoint proceedin
 		if (logger.isDebugEnabled()) {
 			logger.debug("circuitBreaker parameter is null");
 		}
-		return AnnotationExtractor.extract(proceedingJoinPoint.getTarget().getClass(), CircuitBreaker.class);
+		if (proceedingJoinPoint.getTarget() instanceof Proxy) {
+			logger.debug("The circuit breaker annotation is kept on a interface which is acting as a proxy");
+			return extractCircuitBreakerAnnotationFromProxyClass(proceedingJoinPoint);
+		} else {
+			return AnnotationExtractor.extract(proceedingJoinPoint.getTarget().getClass(), CircuitBreaker.class);
+		}
+	}
+
+	@Nullable
+	private CircuitBreaker extractCircuitBreakerAnnotationFromProxyClass(ProceedingJoinPoint proceedingJoinPoint) {
+		if (proceedingJoinPoint.getTarget().getClass().getInterfaces().length > 0) {
+			return AnnotationExtractor.extract(proceedingJoinPoint.getTarget().getClass().getInterfaces()[0], CircuitBreaker.class);
```

Comment:
Does that work, if the Proxy implements multiple interfaces

Reply:
In case of FeignClient the proxy will be implementing only single interface i.e. the original Feign interface. I cannot assume any case where there could be multiple interfaces while using a @FeignClient annotation. 

Though it could be a valid case with some other integrations rather than openfeign. In such case this condition should be converted to a for loop to loop all the interfaces to find the @CircuitBreaker annotation and return from the first match. Please suggest if this case is a valid scenario to cover here, I'll update the code if you suggest.

Reply:
In my opinion we should support proxies which implement multiple interfaces.

Reply:
Done @RobWin, kindly check again.